pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/Common.java

```diff
@@ -172,14 +174,367 @@ public static String javaPrimitiveToObjectType(String primitiveFieldType) {
+			if (f.parent() != null) {
+				final OneOfField oneOfField = f.parent();
+				generatedCodeSoFar += getFieldsHashCode(oneOfField.fields(), generatedCodeSoFar);
+			}
+
+			if (f.optionalValueType()) {
+				generatedCodeSoFar = getOptionalHashCodeGeneration(generatedCodeSoFar, f);
+			}
+			else if (f.repeated()) {
+				generatedCodeSoFar = getRepeatedHashCodeGeneration(generatedCodeSoFar, f);
+			}
+			else if (f.nameCamelFirstLower() != null) {
+				if (f.type() == Field.FieldType.FIXED32 ||
+						f.type() == Field.FieldType.INT32 ||
+						f.type() == Field.FieldType.SFIXED32 ||
+						f.type() == Field.FieldType.SINT32 ||
+						f.type() == Field.FieldType.UINT32) {
+					generatedCodeSoFar += (
+               		"""
+					if ($fieldName != DEFAULT.$fieldName) {
+						result = 31 * result + Integer.hashCode($fieldName);
+					}
+					""").replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.FIXED64 ||
+						f.type() == Field.FieldType.INT64 ||
+						f.type() == Field.FieldType.SFIXED64 ||
+						f.type() == Field.FieldType.SINT64 ||
+						f.type() == Field.FieldType.UINT64) {
+					generatedCodeSoFar += (
+               		"""
+					if ($fieldName != DEFAULT.$fieldName) {
+						result = 31 * result + Long.hashCode($fieldName);
+					}
+					""").replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.ENUM) {
+					generatedCodeSoFar += (
+               		"""
+					if ($fieldName != DEFAULT.$fieldName) {
+						result = 31 * result + $fieldName.hashCode();
+					}
+					""").replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.BOOL) {
+					generatedCodeSoFar += (
+               		"""
+					if ($fieldName != DEFAULT.$fieldName) {
+						result = 31 * result + Boolean.hashCode($fieldName);
+					}
+					""").replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.FLOAT) {
+					generatedCodeSoFar += (
+               		"""
+             		if ($fieldName != DEFAULT.$fieldName) {
+						result = 31 * result + Float.hashCode($fieldName);
+					}
+					""").replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.DOUBLE) {
+					generatedCodeSoFar += (
+               		"""
+					if ($fieldName != DEFAULT.$fieldName) {
+						result = 31 * result + Double.hashCode($fieldName);
+					}
+					""").replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.STRING) {
+					generatedCodeSoFar += (
+               		"""
+					if ($fieldName != DEFAULT.$fieldName) {
+						result = 31 * result + $fieldName.hashCode();
+					}
+					""").replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.BYTES) {
+					generatedCodeSoFar += (
+               		"""
+					if ($fieldName != DEFAULT.$fieldName) {
+						result = 31 * result + ($fieldName == null ? 0 : $fieldName.hashCode());
```

Comment:
Are BYTES fields of type Bytes or byte[]? I assume the former, but if for some reason it is the latter, then Arrays.hashCode() should be used instead.

Reply:
I checked - it's `com.hedera.pbj.runtime.io.buffer.Bytes`